core/state_transition.go

```diff
@@ -315,12 +347,41 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
 	st.refundGas()
 	st.state.AddBalance(st.evm.Coinbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.gasPrice))
 
+	// for all contracts being created as the result of the transaction execution
+	// we build the index for them if multitenancy is enabled
+	if st.evm.SupportsMultitenancy {
+		addresses := evm.CreatedContracts()
+		for _, address := range addresses {
+			log.Debug("Save to extra data",
+				"address", strings.ToLower(address.Hex()),
+				"isPrivate", isPrivate,
+				"parties", managedPartiesInTx)
+			st.evm.StateDB.WriteManagedParties(address, managedPartiesInTx)
+		}
+	}
+
 	if isPrivate {
 		return ret, 0, vmerr != nil, err
 	}
 	return ret, st.gasUsed(), vmerr != nil, err
 }
 
+func containsAll(source, target []string) bool {
+	for _, str := range target {
```

Comment:
MINOR: it looks like we are doing `len(source) x len(target)` complexity. Could we do `len(source) + len(target)` by using a map?

Reply:
changed